events/views.py

```diff
@@ -258,3 +263,107 @@ def check_participant_formset(request, event_instance, meeting_participants_form
             event_instance.delete()
             return False
         return True
+
+
+class MeetingVoteView(TemplateView):
+
+    template_name = 'vote/meeting_vote.html'
+
+    def __init__(self, **kwargs) -> None:
+        self.title = None
+        self.vote_form = None
+        self.VoteFormset = None
+        self.chosen_event = None
+        self.meetings_dates = None
+        self.chosen_meeting_dates = None
+        super().__init__(**kwargs)
+
+    def dispatch(self, request, meeting_id, *args, **kwargs):
+        if not request.user.is_authenticated:
+            return redirect(LOGIN_PAGE)
+
+        try:
+            user_id = request.user
```

Comment:
i think you forgot to use the check_user_auth method  here 😅

Reply:
I didn't understand what you meant,
Did you mean the login required?
because this is a template view there is no login required decorator but there is a dispatch which get the request before it dispatched into the get and post functions and in this way I can redirect the user like I did in lines 282 - 283

Reply:
I didn't understand what you meant,
Did you mean the login required?
because this is a template view there is no login required decorator but there is a dispatch which get the request before it dispatched into the get and post functions and in this way I can redirect the user like I did in lines 282 - 283

Reply:
i dont see where you call this method, ithought maybe you intended to use it here
https://github.com/redhat-beyond/lets-meet/blob/2ad06c8505b44b9b972b1f880ef31ac53a6acd73/events/views.py#L339-L347

Reply:
Right, I have created this function and forgot about it 😄 
I have inserted it at the start of the dispatch, thanks